### PR TITLE
address some FIXME whose associated issues were marked as closed

### PR DESCRIPTION
**Pull request by nivkner**
part of #44366

### CODE REVIEW

File: src/librustc/hir/mod.rs [hunk 1]
```diff
@@ -916,8 +916,7 @@ impl Stmt_ {
     }
 }
 
-// FIXME (pending discussion of #1697, #2178...): local should really be
-// a refinement on pat.
+// FIXME: local should really be a refinement on pat.
```
Comment (petrochenkov) [inline src/librustc/hir/mod.rs]: What this FIXME even mean? I'd rather remove it entirely (and its equivalent in `ast.rs` as well).
#1697 #2178 looks irrelevant.

**Comment by carols10cents** [comment]
Thanks for the PR! We’ll periodically check in on it to make sure that @dtolnay  or someone else from the team reviews it soon.

File: src/libcore/benches/mem.rs [hunk 2]
```diff
@@ -10,7 +10,6 @@
 
 use test::Bencher;
 
-// FIXME #13642 (these benchmarks should be in another place)
```
Comment (dtolnay) [inline src/libcore/benches/mem.rs]: I reopened this issue because I think it was closed prematurely. Please keep this FIXME for now (or fix #13642!).

**Review by dtolnay — COMMENTED** [review]
Thanks so much for tackling this! Looks good to me except for the one that I reopened.

File: src/librustc/benches/lib.rs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+#![deny(warnings)]
+
+#![feature(slice_patterns)]
+#![feature(test)]
+
+extern crate test;
+
+mod dispatch;
+mod match;
```
Comment (dtolnay) [inline src/librustc/benches/lib.rs]:
```
error: expected identifier, found keyword `match`
  --> benches/lib.rs:19:5
   |
19 | mod match;
   |     ^^^^^
```

**Review by dtolnay — COMMENTED** [review]
Thanks for sorting out those benchmarks! I think you need to pick a different module name though.

**Comment by nivkner** [comment]
Actually, can I send another commit to rename the modules to traits and _match to mirror the related modules, like in libcore benches?

**Review by dtolnay — COMMENTED** [review]
I checked off these changes in the checklist in #44366. We are just waiting for this to make it to the top of the [build queue](https://buildbot2.rust-lang.org/homu/queue/rust) and pass tests on all platforms.
